### PR TITLE
[Access] Add options to disable DHT and bitswap reproviding

### PR DESCRIPTION
**Pull request by peterargue**
Backports: https://github.com/onflow/flow-go/pull/5797

This PR makes the following changes, primarily affecting Access and Execution nodes:

1. Disable DHT on the staked network
2. Enable caching for the bitswap datastore

These are targeting issues encountered by nodes running bitswap (EN/AN) that resulted in high memory and CPU utilization.

Note: version updates from backported PR were already applied here: https://github.com/onflow/flow-go/pull/5774

### CODE REVIEW

File: network/p2p/builder/libp2pNodeBuilder.go [hunk 1]
```diff
@@ -463,13 +466,28 @@ func DefaultNodeBuilder(
...
+	if dhtSystemActivation == DhtSystemEnabled {
+		b.SetRoutingSystem(func(ctx context.Context, host host.Host) (routing.Routing, error) {
+			return dht.NewDHT(ctx, host, protocols.FlowDHTProtocolID(b.sporkId), b.logger, b.metricsConfig.Metrics, dht.AsServer())
+		})
+	}
+
+	// bitswap requires a content routing system. this returns a stub instead of a full DHT
+	b.SetRoutingSystem(func(ctx context.Context, host host.Host) (routing.Routing, error) {
+		return none.ConstructNilRouting(ctx, host, nil, nil)
+	})
```
Comment (janezpodhostnik) [inline network/p2p/builder/libp2pNodeBuilder.go:492]: is this supposed to be in an else block?

File: cmd/scaffold.go [hunk 2]
```diff
@@ -2131,12 +2141,12 @@ func DhtSystemActivationStatus(roleStr string) (p2pbuilder.DhtSystemActivation,
...
-		// Only access and execution nodes need to run DHT;
-		// Access nodes and execution nodes need DHT to run a blob service.
-		// Moreover, access nodes run a DHT to let un-staked (public) access nodes find each other on the public network.
-		return p2pbuilder.DhtSystemEnabled, nil
+
+	// Only access and execution nodes need to run DHT; which is used by bitswap.
+	// Access nodes also run a DHT on the public network for peer discovery of un-staked nodes.
+	if role != flow.RoleAccess && role != flow.RoleExecution {
+		return p2pbuilder.DhtSystemDisabled, nil
 	}
```
Comment (janezpodhostnik) [inline cmd/scaffold.go:2149]: Maybe package the check into some function so it can be reused in libp2pNodeBuilder.

**Review by janezpodhostnik — APPROVED** [review]
Just 2 minor comments.

File: network/p2p/builder/libp2pNodeBuilder.go (hunk 1 above)
Comment (peterargue) [reply network/p2p/builder/libp2pNodeBuilder.go]: [d732359](https://github.com/onflow/flow-go/pull/5827/commits/d732359ecb883310869610dd64f2c15019d20234)

**Review by sideninja — APPROVED** [review]

**Review by zhangchiqing — APPROVED** [review]

**Pull request merged**